Predator Tweaks

Therighthon opened the pull request:
- Added Sounds for Predators
- Removed persistence required
- Walk animations faster/slower based on walking speed

Therighthon committed:
Added Crayfish

Added FreshWaterCritters, and Crayfish as the sole variant.

Therighthon committed:
Getting Crayfish to Spawn

Therighthon committed:
Finished Lobster/Crayfish Implementation

Separate Swimming Animation
Both now use the LobsterModel correctly
Aquatic critters flee when attacked

Therighthon committed:
Cougars

Added cougars, a new predator type that will attempt to sneak up on the player before attacking.

Therighthon committed:
Added Panthers

Added panthers as a rare (1 in 10) variant of the cougar

Therighthon committed:
Nocturnal Cougars

Set cougars to be nocturnal

Therighthon committed:
Addressing minor comments

Therighthon committed:
Delete LICENSE.txt

Therighthon committed:
Delete LICENSE.txt

Therighthon committed:
Delete LICENSE.txt

Therighthon committed:
Delete LICENSE.txt

Therighthon committed:
Delete LICENSE.txt

Therighthon committed:
Removed Cougar Class

Therighthon committed:
Big Cats and Animations

Big Cats:
- Lions, Cougars, Panthers, Sabertooths (Saberteeth?)
- Currently behave like normal predators, lower HP than bears but higher attack

Model Changes
- Polar Bears have walk/run/swim/slide animations for movement.
- Big cats have walk/run animations, cougars/panthers still have a sneak animation, but that is currently unused.
- All predators have sleeping animations rather than just poses
- Made alpacas move their heads and necks independently when looking around.
- Predators rear up when attacking

Added taro root, functionally identical to cattail root, but drops from water taro, an emergent found farther south.

Therighthon committed:
Finally learned how to apply the CodeStyle...

...and applied it to all the Java files in the diff, so should be good to go on that front.

Therighthon committed:
Bears and Bugfixes

Addressed comments.

Added Black and Brown bears.

Made predators not despawn when you walk too far away.

eerussianguy committed:
Merge branch '1.18.x' into April2022

# Conflicts:
#	resources/constants.py
#	resources/data.py
#	src/main/java/net/dries007/tfc/client/render/entity/BearRenderer.java

eerussianguy committed:
small predator fixes

eerussianguy committed:
Delete .gitignore

eerussianguy committed:
last minute cleanups

Therighthon committed:
Predator Tune-Ups

- Walk animations now run faster if the predator is moving faster, and slower if it is moving slower.

- Added full sound sets for the existing predators

Therighthon committed:
Merge branch 'May2022' into April2022

Therighthon committed:
Merge pull request #12 from Therighthon/April2022

Merge Predator Updates with May2022

Therighthon committed:
Merge Fixes and Sound Leveling

Made some predator sounds less atrociously loud

eerussianguy commented on src/main/java/net/dries007/tfc/client/model/entity/BearModel.java:
```diff
@@ -162,7 +169,9 @@ else if (predator.isInWater() || (blockStateBelow.getFriction(predator.level, bl
                 SWIM.tick(parts, predator.walkProgress);
             }
-            else if (predator.walkProgress > 0)
+
+            else if (predator.walkProgress > 0 || predator.isMoving())
+
```
This isn't really our code style, was this accidentally done by intelliJ?

Therighthon committed:
Removed Weird Extra Lines

I have no idea why they were there, wasn't auto formatting, probably added them by accident in one model and copied over the code to the others.
